File: src/test/java/aoc2015/day10/Day10Test.java
package aoc2015.day10;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

class Day10Test {

  @ParameterizedTest
  @MethodSource("provideParameters")
  void transformationTest(String input, String expectedOutput) {
    SequenceTransformer transformer = new SequenceTransformer(input);
    transformer.transform();

    assertEquals(expectedOutput, transformer.getSequence());
  }

  private static Stream<Arguments> provideParameters() {
    return Stream.of(
        Arguments.of("1", "11"),
        Arguments.of("11", "21"),
        Arguments.of("21", "1211"),
        Arguments.of("1211", "111221"),
        Arguments.of("111221", "312211")
    );
  }
}

File: src/test/java/aoc2020/day10/Day10Test.kt
package aoc2020.day10

import aoc.loadExample
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource


class Day10Test {

    private val year = 2020
    private var subject: Day10 = Day10()

    @ParameterizedTest
    @CsvSource(
        "day10_1.txt, 35",
        "day10_2.txt, 220"
    )
    fun part1_examples(filename: String, expectedResult: Int) {
        val input = loadExample(year, filename)

        assertThat(subject.part1(input))
            .isEqualTo(expectedResult)
    }

    @ParameterizedTest
    @CsvSource(
        "day10_1.txt, 8",
        "day10_2.txt, 19208"
    )
    fun part2_examples(filename: String, expectedResult: Long) {
        val input = loadExample(year, filename)

        assertThat(subject.part2(input))
            .isEqualTo(expectedResult)
    }
}